### DISCUSSION TITLE
Provide way to configure InitialCredit on creation of RawConsumerConfig

### DISCUSSION THREAD

**Discussion by bSirotkovic** [rabbitmq/rabbitmq-stream-dotnet-client#264] [Ideas]
We've been having serious performance issues with using RMQ Streams when the latency between the broker and the client is high.
We believe the issue is related to the credit flow as described here: https://www.rabbitmq.com/stream.html#consumer-credit-flow

During the creation of `RawConsumerConfig `the subscibe method is called with a hardcoded value of InitialCredit = 2.

`await _client.Subscribe(
                _config,
                InitialCredit,
                consumerProperties,
                async deliver =>
                { ... }
            ).ConfigureAwait(false);`

We would like to have an ability to increase this value in `ConsumerConfig `class

**Comment by Gsantomaggio** [comment]
Do you have the same problem with version `1.2` ? The initial-credits was 10

**Reply by bSirotkovic** [reply]
Version 1.2 was actually in our production so yes. When we imported your source code manually and changed it to 500 we received great increase in performance on staging environment.

**Reply by Gsantomaggio** [reply]
Ok thank you

**Comment by Gsantomaggio** [comment]
@bSirotkovic do you have the opportunity to test https://github.com/rabbitmq/rabbitmq-stream-dotnet-client/pull/265 ?

**Comment by Gsantomaggio** [comment]
@bSirotkovic and @debarisi 


Have you considered using the subentry batch send? So you can compress the messages and receive them in a single chunk send.

like:
```csharp
   var messagesToSend = new List<Message>();
    for (ulong i = 0; i < xxxx ; i++)
    {
        messagesToSend.Add(new Message(new byte[50]));
    }

    await producer.Send(messagesToSend, CompressionType.Gzip).ConfigureAwait(false);
```

**Reply by debarisi** [reply]
@Gsantomaggio ,

We can integrate _Compression_ immediately, but sending messages in bulk will need some code changes.
We will test it tomorrow and return with some numbers as soon as possible.

Thanks for your help.
